> 21:01:54 AssertionError: False is not True : 'frame variable -d run -- d2' returns expected result, got '([NSObject : AnyObject]) d2 = {}'
> ..
> 21:03:44 FAIL: test_dictionary_nsobject_any_object_dsym (lang/swift/variables/dict_nsobj_anyobj/TestSwiftDictionaryNSObjectAnyObject.py)
> 21:03:44 FAIL: test_dictionary_nsobject_any_object_dwarf (lang/swift/variables/dict_nsobj_anyobj/TestSwiftDictionaryNSObjectAnyObject.py)

Hm, the tests might not have picked up the correct swiftc.

--- a/source/Plugins/Language/Swift/SwiftDictionary.cpp
+++ b/source/Plugins/Language/Swift/SwiftDictionary.cpp
@@ -44,6 +44,8 @@ DictionaryConfig::DictionaryConfig()
   // Note: We need have use the old _Tt names here because those are
   // still used to name classes in the ObjC runtime.
 
+  m_nativeStorageRoot_mangled =
+    ConstString("_TtCs21_RawDictionaryStorage");
   m_nativeStorageRoot_demangled =
     ConstString("Swift._RawDictionaryStorage");
 

--- a/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
+++ b/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
@@ -281,7 +281,7 @@ HashedCollectionConfig::CreateEmptyHandler(CompilerType elem_type) const {
 }
 
 ValueObjectSP
-HashedCollectionConfig::SwiftObjectAtAddress(
+HashedCollectionConfig::StorageObjectAtAddress(
     const ExecutionContext &exe_ctx,
     lldb::addr_t address) const {
 
@@ -304,14 +304,15 @@ HashedCollectionConfig::SwiftObjectAtAddress(
   if (error.Fail())
     return nullptr;
 
-  CompilerType anyObject_type = ast_ctx->GetAnyObjectType();
-  if (!anyObject_type)
+  CompilerType rawStorage_type = ast_ctx->GetTypeFromMangledTypename(
+    m_nativeStorageRoot_mangled.GetCString(), error);
+  if (!rawStorage_type.IsValid())
     return nullptr;
 
   lldb::DataBufferSP buffer(
     new lldb_private::DataBufferHeap(&address, sizeof(lldb::addr_t)));
   return ValueObjectConstResult::Create(
-    exe_scope, anyObject_type, ConstString("swift"),
+    exe_scope, rawStorage_type, ConstString("swift"),
     buffer, exe_ctx.GetByteOrder(), exe_ctx.GetAddressByteSize());
 }
 
@@ -356,10 +357,10 @@ HashedCollectionConfig::CreateNativeHandler(
   // runtime class of storage_sp and verify that it's of a known type.
   // If thissuccessful, get the correct key_type and value_type directly
   // from its generic arguments instead of using value_sp.
-  storage_sp = storage_sp->GetQualifiedRepresentationIfAvailable(
+  auto dynamic_storage_sp = storage_sp->GetQualifiedRepresentationIfAvailable(
     lldb::eDynamicCanRunTarget, false);
 
-  auto type = storage_sp->GetCompilerType();
+  auto type = dynamic_storage_sp->GetCompilerType();
 
   auto typeName = type.GetTypeName().GetStringRef();
   if (typeName == m_emptyStorage_demangled.GetStringRef()) {
@@ -370,7 +371,7 @@ HashedCollectionConfig::CreateNativeHandler(
     auto key_type = type.GetGenericArgumentType(0);
     auto value_type = type.GetGenericArgumentType(1);
     if (key_type.IsValid()) {
-      return _CreateNativeHandler(storage_sp, key_type, value_type);
+      return _CreateNativeHandler(dynamic_storage_sp, key_type, value_type);
     }
   }
 
@@ -380,7 +381,7 @@ HashedCollectionConfig::CreateNativeHandler(
   type = value_sp->GetCompilerType();
   CompilerType key_type = type.GetGenericArgumentType(0);
   CompilerType value_type = type.GetGenericArgumentType(1);
-  if (typeName == m_nativeStorageRoot_demangled.GetStringRef()) {
+  if (key_type.IsValid()) {
     return _CreateNativeHandler(storage_sp, key_type, value_type);
   }
   return nullptr;
@@ -619,7 +620,7 @@ HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
   if (valobj_sp->GetObjectRuntimeLanguage() != eLanguageTypeSwift &&
       valobj_sp->IsPointerType()) {
     lldb::addr_t address = valobj_sp->GetPointerValue();
-    if (auto swiftval_sp = SwiftObjectAtAddress(exe_ctx, address))
+    if (auto swiftval_sp = StorageObjectAtAddress(exe_ctx, address))
       valobj_sp = swiftval_sp;
   }
   valobj_sp = valobj_sp->GetQualifiedRepresentationIfAvailable(
@@ -664,7 +665,10 @@ HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
 
   if (masked_storage_location == storage_location) {
     // Native storage
-    return CreateNativeHandler(valobj_sp, bobject_sp);
+    auto storage_sp = StorageObjectAtAddress(exe_ctx, storage_location);
+    if (!storage_sp)
+      return nullptr;
+    return CreateNativeHandler(valobj_sp, storage_sp);
   } else {
     auto cocoa_sp = CocoaObjectAtAddress(exe_ctx, masked_storage_location);
     if (!cocoa_sp)

--- a/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
+++ b/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
@@ -606,10 +606,9 @@ NativeHashedStorageHandler::GetElementAtIndex(size_t idx) {
 HashedStorageHandlerUP
 HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
   static ConstString g_native("native");
-  static ConstString g__variant("_variant"); // Swift 5
-  static ConstString g_cocoa("cocoa"); // Swift 4.2
-  static ConstString g_object("object"); // Swift 5
-  static ConstString g_rawValue("rawValue"); // Swift 5
+  static ConstString g__variant("_variant");
+  static ConstString g_object("object");
+  static ConstString g_rawValue("rawValue");
   static ConstString g__storage("_storage");
 
   Status error;
@@ -645,48 +644,6 @@ HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
 
   ValueObjectSP bobject_sp =
     variant_sp->GetChildAtNamePath({g_object, g_rawValue});
-
-  if (!bobject_sp) { // FIXME: Remove (Legacy Swift 4.2 path)
-    ConstString variant_cs(variant_sp->GetValueAsCString());
-    if (!variant_cs)
-      return nullptr;
-
-    if (g_cocoa == variant_cs) {
-      // it's an NSDictionary/NSSet in disguise
-      static ConstString g_object("object"); // Swift 5
-      static ConstString g_cocoaDictionary("cocoaDictionary"); // Swift 4
-      static ConstString g_cocoaSet("cocoaSet"); // Swift 4
-    
-      ValueObjectSP child_sp =
-        variant_sp->GetChildAtNamePath({g_cocoa, g_object});
-      if (!child_sp) // try Swift 4 name for dictionaries
-        child_sp = variant_sp->GetChildAtNamePath({g_cocoa, g_cocoaDictionary});
-      if (!child_sp) // try Swift 4 name for sets
-        child_sp = variant_sp->GetChildAtNamePath({g_cocoa, g_cocoaSet});
-      if (!child_sp)
-        return nullptr;
-      // child_sp is the _NSDictionary/_NSSet reference.
-      ValueObjectSP ref_sp = child_sp->GetChildAtIndex(0, true); // instance
-      if (!ref_sp)
-        return nullptr;
-
-      uint64_t cocoa_ptr = ref_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
-      if (cocoa_ptr == LLDB_INVALID_ADDRESS)
-        return nullptr;
-      // FIXME: for some reason I need to zero out the MSB; figure out why
-      cocoa_ptr &= 0x00FFFFFFFFFFFFFF;
-
-      auto cocoa_sp = CocoaObjectAtAddress(exe_ctx, cocoa_ptr);
-      if (!cocoa_sp)
-        return nullptr;
-      return CreateCocoaHandler(cocoa_sp);
-    }
-    if (g_native == variant_cs) {
-      auto storage_sp = variant_sp->GetChildAtNamePath({g_native, g__storage});
-      return CreateNativeHandler(valobj_sp, storage_sp);
-    }
-    return nullptr;
-  }
 
   lldb::addr_t storage_location =
     bobject_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);

--- a/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
+++ b/source/Plugins/Language/Swift/SwiftHashedContainer.cpp
@@ -606,9 +606,10 @@ NativeHashedStorageHandler::GetElementAtIndex(size_t idx) {
 HashedStorageHandlerUP
 HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
   static ConstString g_native("native");
-  static ConstString g__variant("_variant");
-  static ConstString g_object("object");
-  static ConstString g_rawValue("rawValue");
+  static ConstString g__variant("_variant"); // Swift 5
+  static ConstString g_cocoa("cocoa"); // Swift 4.2
+  static ConstString g_object("object"); // Swift 5
+  static ConstString g_rawValue("rawValue"); // Swift 5
   static ConstString g__storage("_storage");
 
   Status error;
@@ -644,6 +645,48 @@ HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
 
   ValueObjectSP bobject_sp =
     variant_sp->GetChildAtNamePath({g_object, g_rawValue});
+
+  if (!bobject_sp) { // FIXME: Remove (Legacy Swift 4.2 path)
+    ConstString variant_cs(variant_sp->GetValueAsCString());
+    if (!variant_cs)
+      return nullptr;
+
+    if (g_cocoa == variant_cs) {
+      // it's an NSDictionary/NSSet in disguise
+      static ConstString g_object("object"); // Swift 5
+      static ConstString g_cocoaDictionary("cocoaDictionary"); // Swift 4
+      static ConstString g_cocoaSet("cocoaSet"); // Swift 4
+    
+      ValueObjectSP child_sp =
+        variant_sp->GetChildAtNamePath({g_cocoa, g_object});
+      if (!child_sp) // try Swift 4 name for dictionaries
+        child_sp = variant_sp->GetChildAtNamePath({g_cocoa, g_cocoaDictionary});
+      if (!child_sp) // try Swift 4 name for sets
+        child_sp = variant_sp->GetChildAtNamePath({g_cocoa, g_cocoaSet});
+      if (!child_sp)
+        return nullptr;
+      // child_sp is the _NSDictionary/_NSSet reference.
+      ValueObjectSP ref_sp = child_sp->GetChildAtIndex(0, true); // instance
+      if (!ref_sp)
+        return nullptr;
+
+      uint64_t cocoa_ptr = ref_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+      if (cocoa_ptr == LLDB_INVALID_ADDRESS)
+        return nullptr;
+      // FIXME: for some reason I need to zero out the MSB; figure out why
+      cocoa_ptr &= 0x00FFFFFFFFFFFFFF;
+
+      auto cocoa_sp = CocoaObjectAtAddress(exe_ctx, cocoa_ptr);
+      if (!cocoa_sp)
+        return nullptr;
+      return CreateCocoaHandler(cocoa_sp);
+    }
+    if (g_native == variant_cs) {
+      auto storage_sp = variant_sp->GetChildAtNamePath({g_native, g__storage});
+      return CreateNativeHandler(valobj_sp, storage_sp);
+    }
+    return nullptr;
+  }
 
   lldb::addr_t storage_location =
     bobject_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);

--- a/source/Plugins/Language/Swift/SwiftHashedContainer.h
+++ b/source/Plugins/Language/Swift/SwiftHashedContainer.h
@@ -70,7 +70,7 @@ private:
   CreateCocoaHandler(lldb::ValueObjectSP storage_sp) const;
 
   lldb::ValueObjectSP
-  SwiftObjectAtAddress(
+  StorageObjectAtAddress(
     const ExecutionContext &exe_ctx,
     lldb::addr_t address) const;
 
@@ -96,6 +96,7 @@ protected:
 
   ConstString m_collection_demangledRegex;
   
+  ConstString m_nativeStorageRoot_mangled;
   ConstString m_nativeStorageRoot_demangled;
 
   ConstString m_nativeStorage_mangledRegex_ObjC;

--- a/source/Plugins/Language/Swift/SwiftSet.cpp
+++ b/source/Plugins/Language/Swift/SwiftSet.cpp
@@ -45,6 +45,8 @@ SetConfig::SetConfig()
   // Note: We need to have the old _Tt names here because those are
   // still used to name classes in the ObjC runtime.
 
+  m_nativeStorageRoot_mangled =
+    ConstString("_TtCs14_RawSetStorage");
   m_nativeStorageRoot_demangled =
     ConstString("Swift._RawSetStorage");
 

